src/script/interpreter.cpp

```diff
@@ -1432,6 +1432,46 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
                 }
                 break;
 
+                case OP_SHA256_INIT: // (in -- sha256_ctx)
+                case OP_SHA256_WRITE: // (sha256_ctx in -- sha256_ctx)
+                {
+                    if (stack.size() < (opcode == OP_SHA256_WRITE ? 2 : 1))
+                        return set_error(serror, SCRIPT_ERR_INVALID_STACK_OPERATION);
+
+                    CSHA256 ctx;
+                    if (opcode == OP_SHA256_WRITE) {
+                        if (!ctx.Load(stacktop(-2)))
+                            return set_error(serror, SCRIPT_ERR_UNKNOWN_ERROR);
+                    }
+                    valtype& vch = stacktop(-1);
```

Comment:
I think this line and the next must be inside the `(opcode == OP_SHA256_WRITE)` conditional.

Reply:
While my original proposal was `OP_SHA256_IV`, at the last minute I decided that in practice almost every use of `OP_SHA256_IV` would be immediately followed by `OP_SHA256_WRITE`.  Therefore I've modified my proposal to define `OP_SHA256_INIT` which combines the behaviour of  `OP_SHA256_IV` and `OP_SHA256_WRITE`.  The original behaviour of `OP_SHA256_IV` can be simulated by the script `OP_0 OP_SHA256_INIT`.